python/tchannel/tornado/peer.py

```diff
@@ -169,13 +175,20 @@ def request(self, **kwargs):
         :param service_threshold:
             If ``hostport`` was not specified, this specifies the score
             threshold at or below which peers will be ignored.
+        :param blacklist:
+            Peers on the blacklist won't be chosen.
         """
-        return self.choose(
-            hostport=kwargs.pop('hostport', None),
-            score_threshold=kwargs.pop('score_threshold', None),
-        ).request(**kwargs)
+        peer = self.choose(
+            hostport=kwargs.get('hostport', None),
+            score_threshold=kwargs.get('score_threshold', None),
+            blacklist=kwargs.get('blacklist', None),
```

Comment:
The `**kwargs` interface really needs to be fixed, especially for this and other methods that users will interact with regularly. Save it for a followup diff.